src/battle_ai_main.c

```diff
@@ -1434,6 +1434,8 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
                 return 0;
             if (!ShouldTryOHKO(battlerAtk, battlerDef, aiData->abilities[battlerAtk], aiData->abilities[battlerDef], move))
                 ADJUST_SCORE(-10);
+	    else if (IsDynamaxed(battlerDef))
+		ADJUST_SCORE(-10);
```

Comment:
You're not consistent in adding Dynamax checks as separate conditions (here) or adding them as or conditions to an already existing `if` (see above). I'd choose one and not vary in the same file